### ISSUE TITLE
Training from Scratch Yielding Unusable Results

### ISSUE THREAD

**Issue by BankNatchapol** [myshell-ai/MeloTTS#83]
Hello. I've been working on training a model from scratch using approximately 300 hours of 22kHz audio data. However, I've encountered some problems. In my language, the phenomizer isn't stable, so I've made modifications to the training script to make it character-based instead. Despite these adjustments, the results of my training have been disappointing; the model only seems to produce random noise.

<img width="1009" alt="image" src="https://github.com/myshell-ai/MeloTTS/assets/53829777/463fd44a-a85f-4bbe-b9ee-219f956e40f3">

Below are the losses.  If you've got any ideas or tips on how to rescue my poor model from its noisy fate, I'd be incredibly grateful. 
<img width="231" alt="image" src="https://github.com/myshell-ai/MeloTTS/assets/53829777/40be72fd-5860-4ab8-8e2c-cb56ded8a4f0">
<img width="219" alt="image" src="https://github.com/myshell-ai/MeloTTS/assets/53829777/49c4bc9f-f18a-4c90-bb60-eb40dfa3e120">
<img width="1037" alt="image" src="https://github.com/myshell-ai/MeloTTS/assets/53829777/5c1de2bb-8f61-44f9-bcc0-ef946c440581">
<img width="1042" alt="image" src="https://github.com/myshell-ai/MeloTTS/assets/53829777/d27c3ee5-b095-4495-92c3-7d30a3254e78">

**Comment by jeremy110** [comment]
If you only look at the loss/g/total curve, it seems quite normal. Could you please provide your config.json and also mention what other modifications you have made in the program?

**Comment by BankNatchapol** [comment]
> If you only look at the loss/g/total curve(in my fine-tune case), it seems quite normal. Could you please provide your config.json and also mention what other modifications you have made in the program?

Thanks for replying. Here's my config.json. I modified only on the text parts (g2p, symbols, bert). 
[config_2.json](https://github.com/myshell-ai/MeloTTS/files/14811351/config_2.json)

**Comment by jeremy110** [comment]
I have seen the config.json file, and you didn't add num_languages and num_tones there? Actually, my fine-tuning process was similar to yours. I trained a new language using IPA, and there were some symbols not present in the original config.json, so like you, I replaced some of the symbols with my own. Additionally, I made a mistake initially by directly changing the symbols in config.json, but during training, it reads symbols.py, causing inconsistency between the two. Consequently, the model couldn't understand the sounds properly.

**Comment by ZeaMays14142** [comment]
> Hello. I've been working on training a model from scratch using approximately 300 hours of 22kHz audio data. However, I've encountered some problems. In my language, the phenomizer isn't stable, so I've made modifications to the training script to make it character-based instead. Despite these adjustments, the results of my training have been disappointing; the model only seems to produce random noise.
> 
> <img alt="image" width="1009" src="https://private-user-images.githubusercontent.com/53829777/316862256-463fd44a-a85f-4bbe-b9ee-219f956e40f3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDAxNjE0MzUsIm5iZiI6MTc0MDE2MTEzNSwicGF0aCI6Ii81MzgyOTc3Ny8zMTY4NjIyNTYtNDYzZmQ0NGEtYTg1Zi00YmJlLWI5ZWUtMjE5Zjk1NmU0MGYzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjIxVDE4MDUzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBjOGQzZjQyMDhmYzAwMTAwYWQzZDI4ZWE3NGNmODNiOTg0MWRiZDY0YWE0NjcyOThlYTBiYzQyMGI3MzEwOTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.lQmH-VakBBnzfP4RCwQepkOYMNxPUTgYzs3Xtx646yc">
> Below are the losses. If you've got any ideas or tips on how to rescue my poor model from its noisy fate, I'd be incredibly grateful. <img alt="image" width="231" src="https://private-user-images.githubusercontent.com/53829777/316862427-40be72fd-5860-4ab8-8e2c-cb56ded8a4f0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDAxNjE0MzUsIm5iZiI6MTc0MDE2MTEzNSwicGF0aCI6Ii81MzgyOTc3Ny8zMTY4NjI0MjctNDBiZTcyZmQtNTg2MC00YWI4LThlMmMtY2I1NmRlZDhhNGYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjIxVDE4MDUzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM3OThjYWYxNzViNmExZWNkYmU0Mjk4MTM2NzVmNTAyY2IzOTA0ZjVkZjM2MWUxNmI1MDNjOGU2OTc0OGRmNjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.IoA3ZsFbyeJk0RiBiVvKS2IzoyLURSbh3SRvgn0a2nY"> <img alt="image" width="219" src="https://private-user-images.githubusercontent.com/53829777/316862486-49c4bc9f-f18a-4c90-bb60-eb40dfa3e120.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDAxNjE0MzUsIm5iZiI6MTc0MDE2MTEzNSwicGF0aCI6Ii81MzgyOTc3Ny8zMTY4NjI0ODYtNDljNGJjOWYtZjE4YS00YzkwLWJiNjAtZWI0MGRmYTNlMTIwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjIxVDE4MDUzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZlZjE3ZGZlMDgzZDA1NzhlOGU4YTQxNjQxMTIyOTJiMWFlYzZlYzVlZTNlZDA0ODVlMTNjMzVjNTQ2YTUxOTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.VDvmuNNtAgvlHCmXSCiTUmPzjDL6Frb8F72UOqAKFNY"> <img alt="image" width="1037" src="https://private-user-images.githubusercontent.com/53829777/316862746-5c1de2bb-8f61-44f9-bcc0-ef946c440581.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDAxNjE0MzUsIm5iZiI6MTc0MDE2MTEzNSwicGF0aCI6Ii81MzgyOTc3Ny8zMTY4NjI3NDYtNWMxZGUyYmItOGY2MS00NGY5LWJjYzAtZWY5NDZjNDQwNTgxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjIxVDE4MDUzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFlYmQ2OTQ2M2VjNTNiNjE0ZmI1MWI5ZTY1OTM4NWIxNzc0NmU4NDNiMTYwM2Q2YTA5ZjFhNjI0MzZiNzIxYzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.QKO9PBhH0yNUXnBKdI84xmk2meWO7lJH-roYf9Sa2rY"> <img alt="image" width="1042" src="https://private-user-images.githubusercontent.com/53829777/316862932-d27c3ee5-b095-4495-92c3-7d30a3254e78.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDAxNjE0MzUsIm5iZiI6MTc0MDE2MTEzNSwicGF0aCI6Ii81MzgyOTc3Ny8zMTY4NjI5MzItZDI3YzNlZTUtYjA5NS00NDk1LTkyYzMtN2QzMGEzMjU0ZTc4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjIxVDE4MDUzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEyZDAyZDBkOGUyYzFjMDY4M2Y0ZmJlMmViODJmZmQxZGNjYTA5ODU2NjUyNGNmMWMwYWM3NWM1ZWJhZGI4NmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.h3vZI8LJXiWfCUy314msq1YD56kNMnMjZWvNmZEXdTg">



might i ask why 22KHz and not 44100?
is there a particular reason?

**Comment by Fariq-22** [comment]
what are the modification need to do for train the model from scratch
And how many hours of data required for it?
